core/services/workflows/syncer/handler.go

```diff
@@ -334,103 +334,102 @@ func (w workflowName) Hex() string {
 
-	decodedBinary, config, err := h.workflowArtifactsStore.FetchWorkflowArtifacts(ctx, wfID, payload.BinaryURL, payload.ConfigURL)
+	// If there is not a workflow spec in the database, then fetch artifacts and create it
+	spec, err := h.workflowArtifactsStore.GetWorkflowSpec(ctx, owner, payload.WorkflowName)
 	if err != nil {
-		return err
-	}
-
-	// Always fetch secrets from the SecretsURL
-	var secrets []byte
-	if payload.SecretsURL != "" {
-		secrets, err = h.workflowArtifactsStore.GetSecrets(ctx, payload.SecretsURL, payload.WorkflowID, payload.WorkflowOwner)
+		decodedBinary, config, err := h.workflowArtifactsStore.FetchWorkflowArtifacts(ctx, wfID, payload.BinaryURL, payload.ConfigURL)
 		if err != nil {
-			return fmt.Errorf("failed to get secrets: %w", err)
+			return err
 		}
-	}
 
-	// Calculate the hash of the binary and config files
-	hash, err := pkgworkflows.GenerateWorkflowID(payload.WorkflowOwner, payload.WorkflowName, decodedBinary, config, payload.SecretsURL)
-	if err != nil {
-		return fmt.Errorf("failed to generate workflow id: %w", err)
-	}
-
-	// Pre-check: verify that the workflowID matches; if it doesn’t abort and log an error via Beholder.
-	if !bytes.Equal(hash[:], payload.WorkflowID[:]) {
-		return fmt.Errorf("workflowID mismatch: %x != %x", hash, payload.WorkflowID)
-	}
+		// Always fetch secrets from the SecretsURL
+		var secrets []byte
+		if payload.SecretsURL != "" {
+			secrets, err = h.workflowArtifactsStore.GetSecrets(ctx, payload.SecretsURL, payload.WorkflowID, payload.WorkflowOwner)
+			if err != nil {
+				return fmt.Errorf("failed to get secrets: %w", err)
+			}
+		}
 
-	// Ensure that there is no running workflow engine for the given workflow owner/name.
-	key := EngineRegistryKey{
-		Owner: payload.WorkflowOwner, Name: payload.WorkflowName,
-	}
-	if h.engineRegistry.Contains(key) {
-		return fmt.Errorf("workflow is already running, so not starting it : %s", wfID)
-	}
+		// Calculate the hash of the binary and config files
+		hash, err := pkgworkflows.GenerateWorkflowID(payload.WorkflowOwner, payload.WorkflowName, decodedBinary, config, payload.SecretsURL)
+		if err != nil {
+			return fmt.Errorf("failed to generate workflow id: %w", err)
+		}
 
-	// Save the workflow secrets
-	urlHash, err := h.workflowArtifactsStore.GetSecretsURLHash(payload.WorkflowOwner, []byte(payload.SecretsURL))
-	if err != nil {
-		return fmt.Errorf("failed to get secrets URL hash: %w", err)
-	}
+		// Pre-check: verify that the workflowID matches; if it doesn't abort and log an error via Beholder.
+		if !bytes.Equal(hash[:], payload.WorkflowID[:]) {
+			return fmt.Errorf("workflowID mismatch: %x != %x", hash, payload.WorkflowID)
+		}
 
-	// Create a new entry in the workflow_spec table corresponding for the new workflow, with the contents of the binaryURL + configURL in the table
-	status := job.WorkflowSpecStatusActive
-	if payload.Status == 1 {
-		status = job.WorkflowSpecStatusPaused
-	}
+		// Save the workflow secrets
```

Comment:
Can we correct the comment here? We aren't saving the secrets at all, just fetching them

Reply:
Agree on correcting the comment.

`h.workflowArtifactsStore.GetSecretsURLHash` doesn't even fetch anything, it just hashes. I'd argue it shouldn't be part of the ArtifactsStore, as it's more of a utility. But I'll leave alone for now and just correct the comment.